dashboard/src/modules/components/SearchComponent/index.jsx

```diff
@@ -0,0 +1,39 @@
+import React,{useState} from "react";
+import "./index.css"
+import { InputGroup, TextInput, Button } from "@patternfly/react-core";
+import SearchIcon from "@patternfly/react-icons/dist/esm/icons/search-icon";
+function SearchBox({
+  dataArray,
+  setPublicData,
+  startDate,
+  endDate,
+  setControllerName,
+}) {
+  const [controllerValue, setControllerValue] = useState("");
+  const searchController = () => {
+    let modifiedArray = [];
+    modifiedArray = dataArray.filter((data) => {
```

Comment:
Is there any reason to initialize `modifiedArray` just before setting it?  How about just this:
```
    let modifiedArray = dataArray.filter((data) => {
```